argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,329 @@
+        """Accumulate the statistics for each LiDAR frame.
+
+        Args:
+            gt_fpath: Ground truth file path.
+
+        Returns:
+            cls_stats: Class statistics of shape (N, 8)
+            cls_to_ninst: Mapping of the class names to the number of instances in the ground
+                          truth dataset.
+        """
+        log_id = gt_fpath.parents[1].stem
+        logger.info(f"log_id = {log_id}")
+        ts = gt_fpath.stem.split("_")[-1]
+
+        dt_fpath = self.detection_fpath / f"{log_id}/per_sweep_annotations_amodal/" f"tracked_object_labels_{ts}.json"
+
+        dts = np.array(read_label(str(dt_fpath)))
+        gts = np.array(read_label(str(gt_fpath)))
+
+        cls_stats = defaultdict(list)
+        cls_to_ninst = defaultdict(int)
+        for dt_cls in self.detection_cfg.detection_classes:
+            dt_filtered = filter_instances(
+                dts,
+                dt_cls,
+                filter_metric=self.detection_cfg.detection_metric,
+                max_detection_range=self.detection_cfg.max_detection_range,
+            )
+            gt_filtered = filter_instances(
+                gts,
+                dt_cls,
+                filter_metric=self.detection_cfg.detection_metric,
+                max_detection_range=self.detection_cfg.max_detection_range,
+            )
+
+            logger.info(f"{dt_filtered.shape[0]} detections")
+            logger.info(f"{gt_filtered.shape[0]} ground truth")
+            if dt_filtered.shape[0] > 0:
+                metrics, scores = self.assign(dt_filtered, gt_filtered)
+                cls_stats[dt_cls] = np.hstack((metrics, scores))
+
+            cls_to_ninst[dt_cls] = gt_filtered.shape[0]
+        return cls_stats, cls_to_ninst
+
+    def assign(self, dts: np.ndarray, gts: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+        """Attempt assignment of each detection to a ground truth label.
+
+        Args:
+            dts: Detections of shape (N,).
+            gts: Ground truth labels of shape (M,).
+
+        Returns:
+            metrics: Flags indicating true/false positive and true positive errors (N, 8).
+            scores: Corresponding scores for the true positives/false positives (N,)
+        """
+        n_threshs = len(self.detection_cfg.affinity_threshs)
+        metrics = np.zeros((dts.shape[0], n_threshs + NUM_TP_METRICS))
+
+        # Set the true positive metrics to np.nan since error is undefined on false positives.
+        metrics[:, n_threshs : n_threshs + NUM_TP_METRICS] = np.nan
+        scores, ranks = get_ranks(dts)
+        if gts.shape[0] == 0:
+            return metrics, scores
+
+        affinity_matrix = compute_affinity_matrix(dts, gts, self.detection_cfg.affinity_fn_type)
+
+        # Get the most similar GT label for each detection.
+        gt_matches = np.expand_dims(affinity_matrix[ranks].argmax(axis=1), axis=0)
```

Comment:
i personally think the following would be a bit easier to follow, i find `np.expand_dims, axis=0)` to be a bit wordy:
```python
gt_matches = affinity_matrix[ranks].argmax(axis=1).reshape(1,-1)
```